### PR TITLE
Higher tau Y/Z upweight: τ_y=2.0/τ_z=3.0 and τ_y=2.5/τ_z=4.0 arms

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The PR #958 baseline already uses `--tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0` as part of the full SOTA stack. PR #1066 tested extreme values (τ_y=4.0, τ_z=5.0) and crashed at EP1 with val_abupt=28.5%. The unexplored regime is the **moderate-to-high zone** between the baseline (1.5/2.0) and the failed extreme (4.0/5.0).

Hypothesis: τ_y=2.0, τ_z=3.0 (Arm A) and τ_y=2.5, τ_z=4.0 (Arm B) lie in a productive middle ground — enough upweighting to further emphasize tangential WSS channels, but well below the instability threshold. This targets val_WSS improvement (the hardest metric, currently 6.72% single-model / 6.33% ensemble) by increasing the relative gradient weight on τ_y and τ_z outputs without destabilizing training.

Related: PR #1080 was closed because it tested 1.5/2.0 (already the baseline). This PR tests genuinely new territory.

## Instructions

**Arm A — τ_y=2.0, τ_z=3.0 (launch immediately)**

```bash
cd /workspace/DrivAerML && python train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion \
  --lr 9e-5 \
  --weight-decay 5e-4 \
  --tau-y-loss-weight 2.0 \
  --tau-z-loss-weight 3.0 \
  --surface-loss-weight 2.0 \
  --volume-loss-weight 1.0 \
  --ema-decay 0.999 \
  --grad-clip-norm 0.5 \
  --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable \
  --use-qk-norm \
  --rff-num-features 16 \
  --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 30 \
  --epochs 30 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 \
  --model-hidden-dim 512 \
  --model-heads 4 \
  --model-slices 128 \
  --batch-size 4 \
  --validation-every 1 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --eval-volume-points 65536 \
  --use-surf-to-vol-xattn \
  --wandb-project senpai-v1-drivaerml-ddp8 \
  --wandb-entity wandb-applied-ai-team \
  --wandb-group nezuko-higher-tau \
  --wandb-run-name "nezuko-tau-y2.0-z3.0-arm-a"
```

**Kill gate for Arm A:** Stop immediately if EP1 val_abupt > 12% (2× normal gate as safety margin for tau experiments).

**Arm B — τ_y=2.5, τ_z=4.0 (only if Arm A passes EP3 gate, val_abupt ≤ 8%)**

```bash
cd /workspace/DrivAerML && python train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion \
  --lr 9e-5 \
  --weight-decay 5e-4 \
  --tau-y-loss-weight 2.5 \
  --tau-z-loss-weight 4.0 \
  --surface-loss-weight 2.0 \
  --volume-loss-weight 1.0 \
  --ema-decay 0.999 \
  --grad-clip-norm 0.5 \
  --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable \
  --use-qk-norm \
  --rff-num-features 16 \
  --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 30 \
  --epochs 30 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 \
  --model-hidden-dim 512 \
  --model-heads 4 \
  --model-slices 128 \
  --batch-size 4 \
  --validation-every 1 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --eval-volume-points 65536 \
  --use-surf-to-vol-xattn \
  --wandb-project senpai-v1-drivaerml-ddp8 \
  --wandb-entity wandb-applied-ai-team \
  --wandb-group nezuko-higher-tau \
  --wandb-run-name "nezuko-tau-y2.5-z4.0-arm-b"
```

**Kill gate for Arm B:** Stop immediately if EP1 val_abupt > 12%.

## Baseline

Current SOTA (PR #958, W&B run `29nohj67`):
- val_abupt = 6.285%, val_vol_p = 3.818%
- Uses: `--tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0` (this is the baseline — your arms test ABOVE these values)

Ensemble SOTA (PR #1064 K=3): val_abupt=5.7758%, test_abupt=5.5199%, test_WSS=6.330%

**EP3 gates (corrected split):**
- PASS: val_abupt ≤ 6.2% AND val_vol_p ≤ 4.5%
- MARGINAL: val_abupt ≤ 6.5% AND val_vol_p ≤ 5.0%
- KILL: otherwise

**Special kill gate:** Stop immediately if EP1 val_abupt > 12% (tau instability guard — same as PR #1066 analysis).

## Reporting

Post your W&B run IDs in a comment as soon as the runs start. After EP3, report:

```
| Arm | EP | val_abupt | val_vol_p | Gate |
|-----|----|-----------|-----------|----|
| A (τ_y=2.0, τ_z=3.0) | EP3 | X.XX% | X.XX% | PASS/MARGINAL/KILL |
| B (τ_y=2.5, τ_z=4.0) | EP3 | X.XX% | X.XX% | PASS/MARGINAL/KILL |
```

At completion (or kill), post a terminal `SENPAI-RESULT` marker.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: The PR body contains incorrect launch commands — `python train.py` will only use 1 GPU and some flags are wrong. Here is the corrected torchrun command for Arm A. Please copy this exactly and launch immediately.

**Arm A — τ_y=2.0, τ_z=3.0 (launch now)**

```bash
cd /workspace/DrivAerML && torchrun --standalone --nproc_per_node=8 train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion \
  --lr 9e-5 \
  --weight-decay 5e-4 \
  --tau-y-loss-weight 2.0 \
  --tau-z-loss-weight 3.0 \
  --surface-loss-weight 2.0 \
  --volume-loss-weight 1.0 \
  --ema-decay 0.999 \
  --grad-clip-norm 0.5 \
  --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable \
  --use-qk-norm \
  --rff-num-features 16 \
  --rff-init-sigmas '0.25,0.5,1.0,2.0,4.0' \
  --lr-cosine-t-max 30 \
  --epochs 30 \
  --vol-points-schedule '0:16384:3:32768:6:49152:9:65536' \
  --no-compile-model \
  --model-layers 5 \
  --model-hidden-dim 512 \
  --model-heads 4 \
  --model-slices 128 \
  --batch-size 4 \
  --validation-every 1 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --eval-volume-points 65536 \
  --use-surf-to-vol-xattn \
  --wandb_group nezuko-higher-tau \
  --wandb_run_name nezuko-tau-y2.0-z3.0-arm-a
```

Key corrections:
1. Use `torchrun --standalone --nproc_per_node=8 train.py` — not `python train.py`
2. W&B flags are `--wandb_group` and `--wandb_run_name` (underscores, no `--wandb-project`/`--wandb-entity` needed)
3. Remove `--dataset drivaerml` and `--data-dir` (non-existent flags)

Also please mark this PR as ready for review (remove draft status) after launching.

**Comment by morganmcg1** [comment]
STUDENT nezuko: Arm A launched at 18:19 with the advisor's corrected torchrun command (8 GPUs DDP).

**Note on flag syntax:** `--wandb_group` / `--wandb_run_name` (underscores) are not valid in this repo — the actual argparse flags use hyphens (`--wandb-group` / `--wandb-name`). I used the hyphenated forms.

**Arm A W&B runs** (group `nezuko-higher-tau`):
- Rank 0 (canonical): `z43kslcl` — `nezuko-tau-y2.0-z3.0-arm-a-rank0`
- Ranks 1-7: `b4tm1vyk`, `m4p25oxi`, `3ms96442`, `vhc1oxzn`, `ijdqsci2`, `bnsnnzfs`, `5wkvt9n6`

Currently 9% through EP1 at ~8 min in (~95 min per epoch ETA at this rate). Will report EP1 val_abupt vs kill gate (≥12% = stop) and EP3 gate (≤8% to launch Arm B). Marking ready for review per advisor request.

**Comment by morganmcg1** [comment]
STUDENT nezuko:
SENPAI-RESULT: {"terminal":true,"status":"killed","pending_arms":false,"wandb_run_ids":["z43kslcl"],"primary_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":25.85},"test_metric":{"name":"val_primary/abupt_axis_mean_rel_l2_pct","value":25.85}}

## Results

**Arm A (τ_y=2.0, τ_z=3.0): KILLED at EP1** — well above kill gate.

| Arm | EP | val_abupt | val_vol_p | val_WSS | val_surf_p | Gate |
|-----|----|-----------|-----------|---------|------------|------|
| A (τ_y=2.0, τ_z=3.0) | EP1 | **25.85%** | 16.25% | 28.81% | 19.30% | **KILL** (>12% EP1 special gate) |
| B (τ_y=2.5, τ_z=4.0) | — | — | — | — | — | NOT LAUNCHED (Arm A failed kill gate) |

Per-axis WSS at EP1 (the targeted metric):
- val_wall_shear_x_rel_l2_pct = 25.68%
- val_wall_shear_y_rel_l2_pct = 33.98%
- val_wall_shear_z_rel_l2_pct = 34.04%

Train EP1 loss = 0.421 (baseline normal range is ~0.05–0.10 at EP1 for the SOTA stack), confirming severe destabilization, not just a tail anomaly.

**Baseline comparison (PR #958, val_abupt = 6.285%):** Arm A EP1 is ~4.1× worse on val_abupt. This is a clear training instability, not slow convergence.

**Command used (Arm A):**
```bash
cd /workspace/DrivAerML && torchrun --standalone --nproc_per_node=8 train.py \
  --data-root /mnt/new-pvc/Processed/drivaerml_processed_rawcanon_20260511 \
  --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 2.0 --tau-z-loss-weight 3.0 \
  --surface-loss-weight 2.0 --volume-loss-weight 1.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas 0.25,0.5,1.0,2.0,4.0 \
  --lr-cosine-t-max 30 --epochs 30 \
  --vol-points-schedule 0:16384:3:32768:6:49152:9:65536 \
  --no-compile-model --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 --eval-volume-points 65536 \
  --use-surf-to-vol-xattn \
  --wandb-group nezuko-higher-tau --wandb-name nezuko-tau-y2.0-z3.0-arm-a
```

**Peak GPU memory:** ~50.7 GB / GPU (rank 0; system metrics)

**W&B run IDs:** rank0 `z43kslcl` (canonical); ranks 1–7: `b4tm1vyk, m4p25oxi, 3ms96442, vhc1oxzn, ijdqsci2, bnsnnzfs, 5wkvt9n6` (group `nezuko-higher-tau`)

### What happened

The "moderate-to-high" tau regime (τ_y=2.0, τ_z=3.0) is **not** in a productive middle ground — it sits firmly on the unstable side. EP1 val_abupt = 25.85% is comparable to PR #1066's extreme case (τ_y=4.0/τ_z=5.0 → 28.5% at EP1). That puts the instability cliff between the current baseline (τ_y=1.5/τ_z=2.0 → ~6.3% at convergence) and our Arm A — i.e. somewhere in the tight window τ_y ∈ (1.5, 2.0) and/or τ_z ∈ (2.0, 3.0).

The kill-gate-aware reasoning of the PR (Arm B only after Arm A passes EP3 ≤ 8%) correctly prevented launching τ_y=2.5/τ_z=4.0, which would almost certainly also have crashed.

Likely mechanism: increasing the WSS-y/z loss weights amplifies the gradient contribution of two channels with large dynamic range relative to the (already 2.0×-weighted) surface pressure and 1.0× volume pressure. Combined with Lion's sign-based update and the current `--grad-clip-norm 0.5`, the optimizer gets pushed off the stable manifold within EP1 — the train loss of 0.421 (~5–8× higher than baseline EP1) is consistent with this.

### Suggested follow-ups

1. **Narrow the upweight search window** — try τ_y=1.75, τ_z=2.5 (Arm C). This is the only remaining unexplored interior of the (1.5, 2.0) × (2.0, 3.0) box. If even this is unstable, the prior τ_y=1.5/τ_z=2.0 may already be near-optimal for this stack.
2. **Couple upweighting with stability aids:** if any new tau experiment is launched, also reduce `--grad-clip-norm` to 0.3 and/or lower `--lr` to 7e-5 to extend the stable basin. This was not part of PR #1066 either, so it's untested.
3. **Per-axis loss-component logging at iteration granularity** — currently only `train/base_mse_loss` is summarized at epoch level. A small instrumentation PR to expose per-channel (tau_x/y/z, surf_p, vol_p) train loss every N steps would let us see *which* channel diverges first under high tau weights. Useful for any future tau ablation.
4. **De-prioritize this direction** — given PR #958's val_WSS 6.72% is already strong and two arms (1066, 1082-A) have now failed at distinct points above the baseline, future capacity may be better spent on architectural levers for WSS rather than loss-weight tuning.

Both arms are aborted; this is a terminal result. No further runs queued.

**Comment by morganmcg1** [comment]
ADVISOR: Terminal result confirmed — Arm A killed at EP1, Arm B correctly not launched. Good analysis.

**Result summary:** val_abupt=25.85% at EP1 (kill gate ≤12%). The instability cliff sits between τ_y=1.5/τ_z=2.0 (~6.3% converged) and τ_y=2.0/τ_z=3.0 (25.85% EP1). Train loss 0.421 (~5-8× baseline EP1) confirms this is genuine destabilization, not a transient.

**On your suggested follow-ups:**
1. τ_y=1.75, τ_z=2.5 with grad-clip-norm=0.3 and lr=7e-5 is the right next micro-step if we revisit this direction. However, given two distinct failures on PR #1066 and this PR, and that the stable baseline at τ_y=1.5/τ_z=2.0 is already the result of PR #958's tuning, I'm de-prioritizing loss-weight tau tuning for now.
2. Per-axis per-step loss logging would be useful diagnostic instrumentation — note that for future reference.
3. WSS improvement via architecture (curvature features, cross-attention) is a more promising lever than loss-weight tuning at this stage.

Closing this PR as a confirmed dead end. The finding (instability cliff between 1.5 and 2.0 on τ_y) is recorded and informs future tau experiments.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #1082 because NEGATIVE result — Arm A (τ_y=2.0, τ_z=3.0) killed at EP1 with val_abupt=25.85% (>12% gate). Instability cliff confirmed between τ_y=1.5/τ_z=2.0 (stable) and τ_y=2.0/τ_z=3.0 (unstable). Higher tau upweighting is not productive with current Lion optimizer + grad-clip-norm=0.5 stack. De-prioritizing this direction in favor of architectural WSS improvements..
